test/util/framework/hcp_helper.go

```diff
@@ -374,3 +386,245 @@ func CreateClusterRoleBinding(ctx context.Context, subject string, adminRESTConf
 
 	return nil
 }
+
+// CreateHCPClusterAndWait creates an HCP cluster via direct API call and waits for completion , this is to support lower environments .
+func CreateHCPClusterAndWait(
+	ctx context.Context,
+	hcpClient *hcpsdk20240610preview.HcpOpenShiftClustersClient,
+	resourceGroupName string,
+	hcpClusterName string,
+	cluster hcpsdk20240610preview.HcpOpenShiftCluster,
+	timeout time.Duration,
+) (*hcpsdk20240610preview.HcpOpenShiftCluster, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	poller, err := hcpClient.BeginCreateOrUpdate(ctx, resourceGroupName, hcpClusterName, cluster, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed starting cluster creation %q in resourcegroup=%q: %w", hcpClusterName, resourceGroupName, err)
+	}
+
+	operationResult, err := poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
+		Frequency: StandardPollInterval,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed waiting for cluster=%q in resourcegroup=%q to finish creating: %w", hcpClusterName, resourceGroupName, err)
+	}
+
+	switch m := any(operationResult).(type) {
+	case hcpsdk20240610preview.HcpOpenShiftClustersClientCreateOrUpdateResponse:
+		return &m.HcpOpenShiftCluster, nil
+	default:
+		fmt.Printf("unknown type %T: content=%v", m, spew.Sdump(m))
+		return nil, fmt.Errorf("unknown type %T", m)
+	}
+}
+
+// BuildHCPClusterFromParams builds an HCP cluster object using provided parameters and
+// baked-in defaults that mirror our former module values.
+func BuildHCPClusterFromParams(
+	ctx context.Context,
+	parameters ClusterParams,
+	location string,
+	subscriptionId string,
+	resourceGroupName string,
+	testContext *perItOrDescribeTestContext,
+) (hcpsdk20240610preview.HcpOpenShiftCluster, error) {
+	// Start with a blank cluster and fill in from params/defaults
+	cluster := hcpsdk20240610preview.HcpOpenShiftCluster{
+		Location:   &location,
+		Properties: &hcpsdk20240610preview.HcpOpenShiftClusterProperties{},
+	}
+	cluster.Properties.Platform = &hcpsdk20240610preview.PlatformProfile{}
+
+	// Version and channelGroup (use params; constructors carry defaults)
+	if parameters.OpenshiftVersionId != "" {
+		versionID := parameters.OpenshiftVersionId
+		cluster.Properties.Version = &hcpsdk20240610preview.VersionProfile{ID: &versionID}
+		if cluster.Properties.Version.ChannelGroup == nil {
+			cluster.Properties.Version.ChannelGroup = to.Ptr("stable")
+		}
+	}
+
+	// Managed RG
+	if parameters.ManagedResourceGroupName != "" {
+		cluster.Properties.Platform.ManagedResourceGroup = &parameters.ManagedResourceGroupName
+	}
+
+	// NSG/Subnet/VNET to IDs
+	if parameters.NsgName != "" {
+		nsgID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkSecurityGroups/%s", subscriptionId, resourceGroupName, parameters.NsgName)
```

Comment:
The `.bicep` that created the resource and wrote some `output` parameter for the name, can simply output the ID as well. Please do that and consume the ID as an opaque value - there's no need to do any processing on these values.